.github/workflows/release-ghcr.yaml

```diff
@@ -28,17 +28,18 @@ jobs:
     - name: Waiting for ci/hydra-build:required to complete
       run: |
         while [[ true ]]; do
-          conclusion=$(gh api repos/$GITHUB_REPOSITORY/commits/$GITHUB_SHA/check-runs --jq '.check_runs[] | select(.name == "ci/hydra-build:required") | .conclusion')
+          check_name='ci/hydra-build:required'
+          conclusion=$(gh api "repos/$GITHUB_REPOSITORY/commits/$GITHUB_SHA/check-runs?check_name=$check_name" --jq '.check_runs[].conclusion')
           case "$conclusion" in
             success)
-              echo "ci/hydra-build:required succeeded"
+              echo "$check_name succeeded"
               exit 0;;
-            failure)
-              echo "ci/hydra-build:required failed"
-              exit 1;;
-            *)
-              echo "ci/hydra-build:required pending. Waiting 30s..."
+            '')
```

Comment:
I'm concerned that this won't catch the situation where there is no check to wait for, because the `jq` expression will return an empty string in that situation as well.

To handle this correctly would be a bit tricky, though. If the `jq` expression was `map(.conclusion)` you'd get either `[null]` (for in-progress) or `[]` (for no check). You'd then have to run another `jq` to check whether the array was empty (eg using `length`) and if not, to extract the actual conclusion.

I'm not sure, though, whether it's worth going to this much trouble. Also, it's possible the check doesn't exist because it hasn't started yet.

Reply:
> Also, it's possible the check doesn't exist because it hasn't started yet.

That's right, and therefore I think there is nothing to do differently in the `[null]` vs the `[]` case.

If no check is ever delivered we just run into the regular timeout, but there's nothing we could have done to throw an error earlier, unless we also query the repo settings to see whether the check we expect is configured as required, but I guess that's out of scope for this script, and still would not catch misconfigurations in the remaining CI infra anyway.